File: src/main/java/com/icd/survey/api/controller/SurveyController.java
package com.icd.survey.api.controller;

import com.icd.survey.api.dto.request.SurveyRequest;
import com.icd.survey.api.service.SurveyService;
import com.icd.survey.common.response.ApiResponse;
import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.springframework.http.ResponseEntity;
import org.springframework.validation.annotation.Validated;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@Slf4j
@RestController
@RequestMapping("/survey")
@RequiredArgsConstructor
public class SurveyController {
    private final SurveyService surveyService;

    @PostMapping
    public ResponseEntity<ApiResponse> createSurvey(@Validated @RequestBody SurveyRequest requestDto) {
        return ResponseEntity.ok(ApiResponse.successResponse());
    }
}

File: survey-api/src/main/kotlin/com/ic/surveyapi/form/controller/SurveyController.kt
package com.ic.surveyapi.form.controller

import com.ic.surveyapi.form.controller.dto.SurveyCreateRequest
import com.ic.surveyapi.form.controller.dto.SurveyCreateResponse
import com.ic.surveyapi.form.service.SurveyService
import com.ic.surveyapi.form.service.dto.SurveyCreateRequestDto
import com.ic.surveyapi.util.ObjectMapperUtil
import kotlinx.datetime.toKotlinLocalDateTime
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController
import survey.common.ApiEndpointVersionPrefix

@RestController
@RequestMapping(ApiEndpointVersionPrefix.V1_API_PREFIX)
class SurveyController(
    private val surveyService: SurveyService,
    private val objectMapperUtil: ObjectMapperUtil,
) {
    @PostMapping("/surveys")
    fun createSurveyForms(
        @RequestBody surveyForm: SurveyCreateRequest,
    ): SurveyCreateResponse {
        val surveyEntity =
            surveyService.createSurvey(
                surveyForm = objectMapperUtil.convertClass(value = surveyForm, clazz = SurveyCreateRequestDto::class.java),
            )
        return SurveyCreateResponse(
            uuid = surveyEntity.id,
            title = surveyEntity.title,
            createdDateTime = surveyEntity.createdAt.toKotlinLocalDateTime(),
        )
    }
}
